pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -437,7 +441,11 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 	errCh := parallelize.NewErrorChannel()
 	fh.Parallelizer().Until(ctx, len(c.Victims().Pods), func(index int) {
 		victimPod := c.Victims().Pods[index]
-		if victimPod.DeletionTimestamp != nil {
+		pendingDeletion, err := ev.Handler.PendingDeletion(victimPod)
+		if err != nil {
+			logger.Error(err, "Failed to check if a victim is pending deletion", "preemptor", klog.KObj(pod), "victim", klog.KObj(victimPod))
+		}
+		if victimPod.DeletionTimestamp != nil || pendingDeletion {
```

Comment:
No need to be done in this PR: should we filter out those being-deleted pods before DryRunPreemption? which will (1) improve the DryRunPreemption performance by reducing nodes + we can reduce the duplicated filtering logic in `prepareCandidate` and  `prepareCandidateAsync`